src/core/simulation_store.jl

```diff
@@ -10,32 +10,32 @@ Provides storage of simulation data
 abstract type SimulationStore end
 
 # Required methods:
-# - initialize_stage_storage!
+# - initialize_problem_storage!
 # - read_array
 
-struct SimulationStoreStageRequirements
+struct SimulationStoreProblemRequirements
```

Comment:
As part of this refactor it would be a good idea to unify the structs in this file with the main simulation structs. There is redundant information.

Reply:
Agreed. But we should probably do it in a follow up PR. This is getting big already and there are a lot of test issues that need to be addressed as well.